server/src/test/java/org/spine3/server/storage/StorageFactorySwitchShould.java

```diff
@@ -0,0 +1,172 @@
+ * modification, must retain the above copyright notice and the following
+ * disclaimer.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package org.spine3.server.storage;
+
+import com.google.common.base.Supplier;
+import org.junit.After;
+import org.junit.AfterClass;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.spine3.server.storage.memory.InMemoryStorageFactory;
+import org.spine3.util.Environment;
+
+import java.lang.reflect.Field;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.spy;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.spine3.test.Tests.hasPrivateParameterlessCtor;
+
+/**
+ * @author Alexander Yevsyukov
+ */
+public class StorageFactorySwitchShould {
+
+    // Environment protection START
+    @SuppressWarnings("StaticVariableMayNotBeInitialized")
+    private static Environment storedEnvironment;
+
+    @BeforeClass
+    public static void storeEnvironment() {
+        storedEnvironment = Environment.getInstance().createCopy();
+    }
+
+    @SuppressWarnings("StaticVariableUsedBeforeInitialization")
+    @AfterClass
+    public static void restoreEnvironment() {
+        Environment.getInstance().restoreFrom(storedEnvironment);
+    }
+    // Environment protection END
+
+    private StorageFactorySwitch storageFactorySwitch;
+
+    private final Supplier<StorageFactory> inMemorySupplier = new Supplier<StorageFactory>() {
+        @Override
+        public StorageFactory get() {
+            return InMemoryStorageFactory.getInstance();
+        }
+    };
+
+    @Before
+    public void setUp() throws NoSuchFieldException, IllegalAccessException {
+        storageFactorySwitch = StorageFactorySwitch.instance();
+    }
+
+    @After
+    public void cleanUp() throws NoSuchFieldException, IllegalAccessException {
+        cleanSwitch();
+        Environment.getInstance().reset();
+    }
+
+    /**
+     * Cleans the `storageFactorySwitch` instance by nullifying fields.
```

Comment:
I think, `clear` will be more suitable as an action verb for the intention like this.

Another option to consider is `restore`, if you want to emphasise the returning to the original state.